test/li_test.jl

```diff
@@ -0,0 +1,144 @@
+using SpecialFunctions
+using Base.Test
+
+const SF = SpecialFunctions
+
+
+# golden ratio is used in some tests
+ϕ = (sqrt(5)+1)/2.0
+
+@testset "Polylogarithm Li" begin
+    @testset "Check different input formats" begin
+        @test SF.Li(-1.0, 0.0) ≈ 0.0
+        @test SF.Li(-1, 0.0) ≈ 0.0
+        @test SF.Li(-1.0, Complex(0.0)) ≈ 0.0 + 0.0im # don't seem to need imaginary bit here
+        @test SF.Li(Complex(-1.0), 0.0) ≈ 0.0 + 0.0im
+        @test SF.Li(Complex(-1.0), 0.0) ≈ 0.0
+        @test SF.Li(1, 0) ≈ 0.0
+        x = collect(0.0:0.1:0.9)
+        @test all([SF.Li.(1, x)[i] ≈ SF.Li(1, x[i]) for i=1:length(x)])
+        @test SF.Li(Complex(-1.0), Complex(0.3)) ≈ SF.Li(-1.0, 0.3)
```

Comment:
Yes.  Basically, `≈` with the default tolerance is a sanity check for finding crude bugs.  If you really want to carefully check the accuracy, however, you need to look at both the real and imaginary parts separately, and with a lower tolerance, which is why I defined `≅`.